warehouse/admin/templates/admin/organizations/detail.html

```diff
@@ -86,51 +86,76 @@ <h4 class="modal-title" id="renameModalLabel">
-            <div class="form-group">
-              <label class="col-sm-5 control-label">
-                Organization Account Name
-              </label>
-              <div class="col-sm-7">
-                <input class="form-control" value="{{ organization.name }}" readonly>
+          <form method="POST" action="{{ request.route_path('admin.organization.detail', organization_id=organization.id) }}">
+            <input name="csrf_token" type="hidden" value="{{ request.session.get_csrf_token() }}">
+            <div class="form-horizontal">
+              <div class="form-group">
+                <label class="col-sm-5 control-label">
+                  Organization Account Name
+                </label>
+                <div class="col-sm-7">
+                  <input class="form-control" value="{{ organization.name }}" readonly>
+                  <small class="form-text text-muted">Account name cannot be changed here. Use the Rename action instead.</small>
+                </div>
               </div>
-            </div>
-            <div class="form-group">
-              <label class="col-sm-5 control-label">
-                Organization Name
-              </label>
-              <div class="col-sm-7">
-                <input class="form-control" value="{{ organization.display_name }}" readonly>
+              <div class="form-group">
+                <label for="display_name" class="col-sm-5 control-label">
+                  Organization Display Name
+                </label>
+                <div class="col-sm-7">
+                  {{ form.display_name(class="form-control", value=organization.display_name, disabled=(not request.has_permission(Permissions.AdminOrganizationsWrite))) }}
+                  {% if form.display_name.errors %}
+                    <div class="text-danger">{{ form.display_name.errors[0] }}</div>
+                  {% endif %}
+                </div>
               </div>
-            </div>
-            <div class="form-group">
-              <label class="col-sm-5 control-label">
-                Organization URL <i class="fa-solid fa-arrow-up-right-from-square"></i>
-              </label>
-              <div class="col-sm-7">
-                <a href="{{ organization.link_url }}" rel="nofollow noopener noreferrer" target="_blank"><input class="form-control" value="{{ organization.link_url }}" readonly></a>
+              <div class="form-group">
+                <label for="link_url" class="col-sm-5 control-label">
+                  Organization URL <i class="fa-solid fa-arrow-up-right-from-square"></i>
+                </label>
+                <div class="col-sm-7">
+                  {{ form.link_url(class="form-control", value=organization.link_url, disabled=(not request.has_permission(Permissions.AdminOrganizationsWrite))) }}
+                  {% if form.link_url.errors %}
+                    <div class="text-danger">{{ form.link_url.errors[0] }}</div>
+                  {% endif %}
```

Comment:
With this change we lose the ability to click an a href to visit the URL.

suggestion: move the href to the `label`'s icon right above the form

Reply:
will have the robot fix this.

Reply:
f5dfb10ac